src/com/palmergames/bukkit/towny/utils/PlayerCacheUtil.java

```diff
@@ -411,6 +411,25 @@ else if (CombatUtil.isEnemy(resident.getTown(), town)) {
 	 * @return true if allowed.
 	 */
 	private static boolean getPermission(Player player, TownBlockStatus status, WorldCoord pos, Material material, TownyPermission.ActionType action) {
+		// Allow admins to have ALL permissions
+		TownyUniverse townyUniverse = TownyUniverse.getInstance();
+		if (townyUniverse.getPermissionSource().isTownyAdmin(player))
+			return true;
+
+		//If town is bankrupt, nobody can build
+		TownBlock townBlock = null;
+		Town targetTown = null;
+		if(TownySettings.isTownBankruptsyEnabled() && action == ActionType.BUILD) {
+			try {
+				townBlock = pos.getTownBlock();
+				targetTown = townBlock.getTown();
+				if(targetTown.isBankrupt())  {
+					cacheBlockErrMsg(player, TownySettings.getLangString("msg_err_bankrupt_town_cannot_build"));
+					return false;
+				}
+			} catch (NotRegisteredException e) {
+			}
+		}

```

Comment:
This will have to be redone, probably using a new TownBlockStatus.BANKRUPT.

Reply:
- I tried this initially
- There is an issue though :
1. By using TownBlockStatus.BANKRUPT, this seems to remove information about whether the player is an Ally, Resident, Outsider etc.
2. This is no problem for a 'build' action.  (because we just cancel knowing the town is bankrupt)
3. However for other types of actions which we want to allow (e.g. 'destroy', 'switch'), the code cannot proceed because the information about Ally, Resident etc., has been removed.